src/world/Server/Definitions.h

```diff
@@ -1,36 +1,39 @@
 /*
- * AscEmu Framework based on ArcEmu MMORPG Server
- * Copyright (c) 2014-2019 AscEmu Team <http://www.ascemu.org>
- * Copyright (C) 2008-2012 ArcEmu Team <http://www.ArcEmu.org/>
- * Copyright (C) 2005-2007 Ascent Team
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU Affero General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU Affero General Public License for more details.
- *
- * You should have received a copy of the GNU Affero General Public License
- * along with this program. If not, see <http://www.gnu.org/licenses/>.
- */
+Copyright (c) 2014-2019 AscEmu Team <http://www.ascemu.org>
+This file is released under the MIT license. See README-MIT for more information.
+*/
 
-#ifndef WOWSERVER_DEFINITION_H
-#define WOWSERVER_DEFINITION_H
+#pragma once
 
+//////////////////////////////////////////////////////////////////////////////////////////
+// NUM_MAPS
+//
+// \param Max maps 
+//
+// Vanilla = untested
+// The Burning Crusade = untested
+// Wrath of the Lich King = untested
+// Cataclysm = untested
+// Mists of Pandaria = untested
+// Warlords of Draenor = untested
+// Legion = untested
+//
+//////////////////////////////////////////////////////////////////////////////////////////
 
-#if VERSION_STRING < Cata
-const unsigned NUM_MAPS = 800;
-#else
-const unsigned NUM_MAPS = 975; // at least Darkmoon Faire
+#if VERSION_STRING == Classic
+    #define NUM_MAPS 600
+#elif VERSION_STRING == TBC
+    #define NUM_MAPS 600
+#elif VERSION_STRING == WotLK
+    #define NUM_MAPS 800
+#elif VERSION_STRING == Cata
+    #define NUM_MAPS 975
+#elif VERSION_STRING == Mop
+    #define NUM_MAPS 975
```

Comment:
These numbers.... can you check them an implement the correct max map ids instead of som guessed values?

Reply:
// Vanilla = ~
// The Burning Crusade = 598
// Wrath of the Lich King = 724
// Cataclysm = 724
// Mists of Pandaria = ~

perhaps clients of different locations have different number of areas.
or data needed by brothers from another source.